sources/packages/api/src/services/application/application.service.ts

```diff
@@ -1003,4 +1007,94 @@ export class ApplicationService extends RecordDataModelService<Application> {
     application.coeStatus = COEStatus.declined;
     return this.repo.save(application);
   }
+
+  /**
+   * Associates an MSFAA number to the application checking
+   * whatever is needed to create a new MSFAA or use an
+   * existing one instead.
+   * @param applicationId application id to receive an MSFAA.
+   */
+  async associateMSFAANumber(applicationId: number): Promise<Application> {
+    const application = await this.repo.findOne(applicationId, {
+      relations: ["offering"],
+    });
+    if (!application) {
+      throw new CustomNamedError(
+        "Student Application not found.",
+        APPLICATION_NOT_FOUND,
+      );
+    }
+
+    if (application.applicationStatus !== ApplicationStatus.assessment) {
+      throw new CustomNamedError(
+        `Student Application is not in the expected status. The application must be in application status '${ApplicationStatus.assessment}' to an MSFAA number be assigned.`,
```

Comment:
'an MSFAA' , please change to 'a MSFAA'

Reply:
I believe that "a MSFAA" is not correct base on the below explanation...


We listen to the beginning sound of the word following the indefinite articles a and an. If it sounds like a consonant, we use a. If it sounds like a vowel, we use an. Some letters can sound like a vowel or consonant. When the following word is an abbreviation, listen to how the abbreviation is spoken. We hear a vowel sound at the beginning of MBA (em-be-a) and a consonant sound at the beginning of NASA.
Incorrect
a hour, a MBA, a orange, an horse, an university
All of the following examples are correct. Listen to the beginning sound of the word, abbreviation, or number that follows the indefinite article.
Correct
a horse, an hour, a HUD loan,
Correct
a historic site, an honor, an html code,
Correct
an orange, a one-horse town, a $20 bill,
Correct
a pencil, a big elephant, an egotistical person,
Correct
an MBA, a US ambassador, an iPhone,
Correct
an apple, an egg, an inexpensive car,
Correct
a unicorn, an umbrella, a used car